src/Interpreters/Streaming/CalculateDataStreamSemantic.h

```diff
@@ -33,6 +42,7 @@ DataStreamSemanticPair calculateDataStreamSemantic(
     DataStreamSemanticEx left_input_data_stream_semantic,
     std::optional<DataStreamSemanticEx> right_input_data_stream_semantic,
     std::optional<std::pair<JoinKind, JoinStrictness>> kind_and_strictness,
+    std::optional<bool> right_input_is_streaming,
```

Comment:
No need to add this param. It is better to put this variable into 'right_input_data_stream_semantic'